playground/backend/internal/cache/redis_cache.go

```diff
@@ -0,0 +1,148 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	pb "beam.apache.org/playground/backend/internal/api"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/go-redis/redis/v8"
+	"github.com/google/uuid"
+	"time"
+)
+
+const defaultExpirationTime = time.Minute * 15
+
+type RedisCache struct {
+	redisClient *redis.Client
+}
+
+// newRedisCache returns Redis implementation of Cache interface.
+// In case of problem with connection to Redis returns error.
+func newRedisCache(addr string) (*RedisCache, error) {
+	// ctx should be as a parameter
+	ctx := context.Background()
+
+	rc := RedisCache{redisClient: redis.NewClient(&redis.Options{Addr: addr})}
+	_, err := rc.redisClient.Ping(ctx).Result()
+	if err != nil {
+		return nil, err
+	}
+	return &rc, nil
+}
+
+func (rc *RedisCache) GetValue(pipelineId uuid.UUID, subKey SubKey) (interface{}, error) {
+	// ctx should be as a parameter
+	ctx := context.Background()
+
+	switch subKey {
+	case SubKey_Status:
+		var result pb.Status
+		err := get(ctx, rc.redisClient, pipelineId.String(), subKey, &result)
+		if err != nil {
+			return nil, err
+		}
+		return result, nil
+	default:
+		var result string
+		err := get(ctx, rc.redisClient, pipelineId.String(), subKey, &result)
```

Comment:
Can we only make a ```result``` variable in the switch and call the get method once for all cases? looks like the code is completely the same here and above